packages/core/src/Site/index.js

```diff
@@ -844,9 +844,39 @@ class Site {
   }
 
   async reloadSiteConfig() {
+    const oldSiteConfig = this.siteConfig;
     const oldAddressablePages = this.addressablePages.slice();
     const oldPagesSrc = oldAddressablePages.map(page => page.src);
     await this.readSiteConfig();
+    await this.handleIgnoreReload(oldSiteConfig.ignore);
+    await this.handlePageReload(oldAddressablePages, oldPagesSrc, this.isGlobalConfigModified(oldSiteConfig));
+    await this.handleStyleReload(oldSiteConfig.style);
+  }
+
+  /**
+   * Checks if any of the global attributes of site.json is modified
+   */
+  isGlobalConfigModified(oldSiteConfig) {
+    return !_.isEqual(oldSiteConfig.faviconPath, this.siteConfig.faviconPath)
+        || !_.isEqual(oldSiteConfig.titlePrefix, this.siteConfig.titlePrefix)
+        || !_.isEqual(oldSiteConfig.style, this.siteConfig.style)
+        || !_.isEqual(oldSiteConfig.externalScripts, this.siteConfig.externalScripts)
+        || !_.isEqual(oldSiteConfig.globalOverride, this.siteConfig.globalOverride)
+        || !_.isEqual(oldSiteConfig.ignore, this.siteConfig.ignore)
```

Comment:
👀 what's this for?

Reply:
This is to check if any of these configs have changed and if they have changed, would require a rebuild of all pages.

Reply:
(`ignore` in particular) Shouldn't be necessary I think; `ignore` is only used down the line for link validation warnings

Reply:
Ahh ok. I was under the impression that if a page is dependent on an asset that is later added to `ignore`, then that entire page will have to be rebuilt. Might have misunderstood the functionality of `ignore` here 😅